subsys/bluetooth/services/fast_pair/fp_crypto/fp_crypto_tinycrypt.c

```diff
@@ -27,7 +28,30 @@ int fp_crypto_sha256(uint8_t *out, const uint8_t *in, size_t data_len)
 	return 0;
 }
 
-int fp_crypto_aes128_encrypt(uint8_t *out, const uint8_t *in, const uint8_t *k)
+int fp_crypto_hmac_sha256(uint8_t *out,
+			  const uint8_t *in,
+			  size_t data_len,
+			  const uint8_t *hmac_key,
+			  size_t hmac_key_len)
+{
+	struct tc_hmac_state_struct s;
+
+	if (tc_hmac_set_key(&s, hmac_key, hmac_key_len) != TC_CRYPTO_SUCCESS) {
+		return -EINVAL;
```

Comment:
Maybe -EFAULT would be more suitable here?

Reply:
See the API documentation

```
/**
 *  @brief HMAC set key procedure
 *  Configures ctx to use key
 *  @return returns TC_CRYPTO_SUCCESS (1)
 *          returns TC_CRYPTO_FAIL (0) if
 *                ctx == NULL or
 *                key == NULL or
 *                key_size == 0
 * @param ctx IN/OUT -- the struct tc_hmac_state_struct to initial
 * @param key IN -- the HMAC key to configure
 * @param key_size IN -- the HMAC key size
 */
```

The error only happens if the input information is incorrect. That's why I think that the -EINVAL is a better fit